internal/controller/kyma/controller.go

```diff
@@ -294,6 +294,11 @@ func (r *Reconciler) reconcile(ctx context.Context, kyma *v1beta2.Kyma) (ctrl.Re
 		return r.requeueWithError(ctx, kyma, fmt.Errorf("could not synchronize remote kyma status: %w", err))
 	}
 
+	if kyma.Status.State == shared.StateError {
+		// Requeue with a new Error in case of Kyma error state, to enable rate limiting for that error.
+		return ctrl.Result{}, ErrKymaInErrorState
+	}
+
 	return res, err
```

Comment:
```suggestion
	return res, nil
```

Please double check but if I am not mistaken, `err` will always be nil here. Let's make it explicit